### PR TITLE
Validate integers fit in cells

### PR DESCRIPTION
**Pull request by jrtc27**
Currently we silently truncate integers to fit into cells, and integers that don't even fit in an unsigned long long are first clamped to ULLONG_MAX by virtue of strtoull. This PR fixes both of those issues, with one documented catch.

### CODE REVIEW

**Comment by jrtc27** [comment]
Ping?

**Comment by davidchisnall** [comment]
Sorry, CI was failing and I didn't want to merge things untested.  CI should be fixed now, please can you rebase the branch so that it is tested.  Happy to merge once it passes.

File: fdt.cc [hunk 1]
```diff
@@ -335,10 +335,27 @@ property::parse_cells(text_input_buffer &input, int cell_size)
...
 				valid = false;
 				return;
 			}
+			// FIXME: No sign information available, so cannot
+			// distinguish small negative values from large
+			// positive ones, and thus we have to conservatively
+			// permit anything that looks like a sign-extended
+			// negative integer.
+			if (cell_size < 64 && val >= (1ull << cell_size) &&
+			    (val | ((1ull << (cell_size - 1)) - 1)) != ULLONG_MAX)
```
Comment (davidchisnall) [inline fdt.cc:351]: This is causing CI failures.  The correct spelling is `std::numeric_limits<unsigned long long>::max()`.

Comment (jrtc27) [reply fdt.cc:352]: Oh I can do the verbose C++ version if you'd rather than instead of climits

Comment (davidchisnall) [reply fdt.cc:352]: Yes, please use the idiomatic C++ version rather than using C macros.

**Pull request merged**